crates/driver/src/domain/competition/mod.rs

```diff
@@ -312,19 +313,25 @@ impl Competition {
         &self,
         settlement: &Settlement,
     ) -> Result<(), infra::simulator::Error> {
-        self.simulator
-            .gas(eth::Tx {
-                from: self.solver.address(),
-                to: settlement.solver(),
-                value: eth::Ether(0.into()),
-                input: crate::util::Bytes(settlement.calldata(
-                    self.eth.contracts().settlement(),
-                    settlement::Internalization::Enable,
-                )),
-                access_list: settlement.access_list.clone(),
-            })
-            .await
-            .map(|_| ())
+        let tx = eth::Tx {
+            from: self.solver.address(),
+            to: settlement.solver(),
+            value: eth::Ether(0.into()),
+            input: crate::util::Bytes(settlement.calldata(
+                self.eth.contracts().settlement(),
+                settlement::Internalization::Enable,
+            )),
+            access_list: settlement.access_list.clone(),
+        };
+        let gas_needed_for_tx = self.simulator.gas(&tx).await?;
+        if gas_needed_for_tx > settlement.gas.limit {
```

Comment:
Why do we need to check this in our code? Isn't there a more elegant way to let simulator handle gas limit for us? tenderly, web3 and enso all should have a support for this.

This would require to add optional field `eth::Tx::gas` and forward the `settlement.gas.limit`

Reply:
@Mateo-mro initially wanted to do it in the simulator as well but I recommended to do it in the domain logic. My reasoning was that overall if this is just intended to be a economic viability check it's more useful to know whether or not the solution works at all (e.g. to detect issues with the encoding) and only enforce the gas limit afterwards.

Reply:
Thanks @MartinquaXD for the answer! :pray: 

> This would require to add optional field eth::Tx::gas and forward the settlement.gas.limit

@sunce86  what do you exactly mean? :thinking:

Reply:
> @sunce86 what do you exactly mean? 

Just that the eth::Tx struct currently does not support providing gas limit field.

@MartinquaXD makes sense 👍